src/components/offer-card/offer-card.tsx

```diff
@@ -0,0 +1,85 @@
+import { IOffer } from '../../mocks/offers';
+
+//Так себе название, потому что же там второй пропс есть - ключ
+type offerCardPropType = {
+	item: IOffer;
+}
+
+type keyPropType = {
+	key: string;
+}
+
+//Тупое название, тупая ситуация. Если я не задам здесь тип, то ниже не даёт обратиться к полю по [item.rating]
+type widthByRatingType = {
+	[key: number]: string;
+}
+
+const WidthByRating: widthByRatingType = {
+	0: '0',
+	1: '20%',
+	2: '40%',
+	3: '60%',
+	4: '80%',
+	5: '100%'
+} as const;
+
+//Здесь запутался с фигурными скобками. Почему их нужно ставить?
+function OfferCard({ item }: offerCardPropType, { key }: keyPropType): JSX.Element {
+	const favoriteButtonClass = item.isFavorite ? 'place-card__bookmark-button place-card__bookmark-button--active button' : 'place-card__bookmark-button button';
```

Comment:
У вас есть вспомогательный пакет для такого
```suggestion

	const favoriteButtonClass = classNames('place-card__bookmark-button', {
		'place-card__bookmark-button--active': item.isFavorite
	}, 'button');
```